proto/tendermint/types/params.proto

```diff
@@ -27,6 +27,9 @@ message BlockParams {
   // Max gas per block.
   // Note: must be greater or equal to -1
   int64 max_gas = 2;
+  // Indicates if CheckTx should be called
+  // for all transactions after a block is executed
+  bool recheck_tx = 3;
```

Comment:
What's the rationale behind having this as part of `BlockParams` as opposed to, e.g., embedding it in a dedicated `MempoolParams` struct inside `ConsensusParams`? The `BlockParams` message description probably needs to be updated if we keep it here, as `recheck_tx` doesn't seem to have anything to do with block size limits, right?

Reply:
I don't have a _very_ strong opinion on this one, but I do think that at the very least the API docs need to be clearer. If we anticipate that there will be more knobs and dials for the mempool in future, we may want to rather introduce a `MempoolParams` struct.

A longer-term alternative may be to just flatten the consensus params entirely :slightly_smiling_face: Are the groupings computationally meaningful, or are they just there for developer ergonomics? If it's the latter, we could just have one big struct with spacing/comments in between to logically group them in the code.

Reply:
I think this is a matter of ergonomics. Only the entire consensus params are used for hashing. I don't mind adding the `MempoolParams` or maybe call it `TxParams`